File: source-connector/src/main/java/sourceconnector/repository/file/decompressor/ZipDecompressor.java
package sourceconnector.repository.file.decompressor;

import offsetmanager.domain.file.FileKey;

import java.io.IOException;
import java.io.InputStream;
import java.util.zip.ZipEntry;
import java.util.zip.ZipInputStream;

public class ZipDecompressor implements Decompressor {
  @Override
  public boolean supports(FileKey fileKey) {
    return fileKey.get().endsWith(".zip");
  }

  @Override
  public InputStream decompress(InputStream in) throws IOException {
    ZipInputStream zipInputStream = new ZipInputStream(in);
    ZipEntry zipEntry = zipInputStream.getNextEntry();
    if (zipEntry == null) {
      throw new IOException("Empty zip file");
    }
    return zipInputStream;
  }
}

File: source-connector/src/main/kotlin/sourceconnector/repository/file/decompressor/ZipDecompressor.kt
package sourceconnector.repository.file.decompressor

import offsetmanager.domain.file.FileKey
import java.io.IOException
import java.io.InputStream
import java.util.zip.ZipInputStream

class ZipDecompressor : Decompressor {
  override fun supports(fileKey: FileKey): Boolean {
    return fileKey.get().endsWith(".zip")
  }

  @Throws(IOException::class)
  override fun decompress(`in`: InputStream): InputStream {
    val zipInputStream = ZipInputStream(`in`)
    zipInputStream.getNextEntry() ?: throw IOException("Empty zip file")
    return zipInputStream
  }
}
